SLING-11685 use feature launcher assembly to launch

enapps-enorman opened the pull request:
unpack the feature launcher assembly and use the contained script

enapps-enorman committed:
SLING-11158 use feature launcher assembly to launch

unpack the feature launcher assembly and use the contained script

enapps-enorman committed:
SLING-11158 test using feature launcher assembly to launch

rombert commented on src/main/java/org/apache/sling/maven/feature/launcher/StartMojo.java:
```diff
@@ -124,24 +171,58 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-                    for (String vmOption : vmOptions) {
-                        if (vmOption != null && !vmOption.isEmpty()) {
-                            args.add(vmOption);
+                if (useAssembly) {
```
I have not checked, but I assume that the launcher script consults the `JAVA_HOME` env var, right?

rombert commented on src/main/java/org/apache/sling/maven/feature/launcher/StartMojo.java:
```diff
@@ -124,24 +171,58 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-                    for (String vmOption : vmOptions) {
-                        if (vmOption != null && !vmOption.isEmpty()) {
-                            args.add(vmOption);
+                if (useAssembly) {
```
I have not checked, but I assume that the launcher script consults the `JAVA_HOME` env var, right?

enapps-enorman replied:
Yes the launcher script does have some logic for finding where the java executable is.  But thinking more about this more, it probably wouldn't hurt to set the JAVA_HOME env variable here to deal with things like SLING-9843

I''ll make that change.

rombert commented on src/it/simple-it/pom.xml:
```diff
@@ -101,6 +101,7 @@
                 <artifactId>@project.artifactId@</artifactId>
                 <version>@project.version@</version>
                 <configuration>
+                    <featureLauncherVersion>1.1.26</featureLauncherVersion>
```
It would be good to add a comment saying why we pin the version here.

rombert commented on src/it/simple-it/pom.xml:
```diff
@@ -101,6 +101,7 @@
                 <artifactId>@project.artifactId@</artifactId>
                 <version>@project.version@</version>
                 <configuration>
+                    <featureLauncherVersion>1.1.26</featureLauncherVersion>
```
It would be good to add a comment saying why we pin the version here.

enapps-enorman replied:
sure.  I'll add a comment

enapps-enorman committed:
SLING-11158 cleanup

set the JAVA_HOME env variable for SLING-9843
add comment to simple-it for why the featureLauncherVersion is pinned
rename simple-for-SLING-1095600-it test package name so it is unique

enapps-enorman committed:
Merge branch 'master' into issue/SLING-11158

enapps-enorman committed:
SLING-11158 fix missing dependency scope

enapps-enorman committed:
SLING-11158 no need for the vmOptions null check anymore
